### ISSUE TITLE
withObjects parameters of lwm2m_update_registration is not used

### ISSUE THREAD

**Issue by sbernard31** [eclipse/wakaama#231]
Here is the public API description : 

```c
// send a registration update to the server specified by the server short identifier
// or all if the ID is 0.
// If withObjects is true, the registration update contains the object list.
int lwm2m_update_registration(lwm2m_context_t * contextP, uint16_t shortServerID, bool withObjects);
```

When we call this function,  server `status` is set to `STATE_REG_UPDATE_NEEDED`. Then in `registration_step()`, we see that we always send registration update with object :
```c
....
        case STATE_REG_UPDATE_NEEDED:
            // here withObject is set to true (3rd parameters)
            prv_updateRegistration(contextP, targetP, true); 
            break;
....
```

**Comment by fdur** [comment]
See #221 :)

**Comment by dnav** [comment]
Closing as duplicate of #221 (which now has a proposed fix #234 )
